app/services/reindex_service.rb

```diff
@@ -0,0 +1,24 @@
+class ReindexService
+  attr_reader :logger, :index_alias, :indexer, :alias_updater, :index_deleter
+
+  def initialize(args)
+    @indexer = args[:indexer]
+    @alias_updater = args[:alias_updater]
+    @index_deleter = args[:index_deleter]
+    @logger = args[:logger]
+  end
+
+  def run
+    logger.info "Indexing #{published_datasets_count} datasets"
+    indexer.run
+    alias_updater.run
+    index_deleter.run
+    logger.info 'Import complete'
```

Comment:
maybe something more consistent with the phrasing on line 12? e.g. `Indexing complete`. Also, it's the reindex_service, so why not say `Reindexing complete` ?